mesh/meshdb.go

```diff
@@ -127,36 +126,49 @@ func (m *MeshDB) LayerBlockIds(index types.LayerID) ([]types.BlockID, error) {
 	return blockids, nil
 }
 
-func (mc *MeshDB) ForBlockInView(view map[types.BlockID]struct{}, layer types.LayerID, blockHandler func(block *types.BlockHeader) error) error {
-	stack := list.New()
-	for b := range view {
-		stack.PushFront(b)
-	}
-	set := make(map[types.BlockID]struct{})
-	for b := stack.Front(); b != nil; b = stack.Front() {
-		a := stack.Remove(stack.Front()).(types.BlockID)
+type blockIdQueue struct {
+	l []types.BlockID
+}
+
+func (s *blockIdQueue) Push(b types.BlockID) {
```

Comment:
already said that i don't think this is a good idea,
if you do this please change all uses of list with your implementation

Reply:
reverted this change d73734e4